addons/website/static/tests/tours/carousel_content_removal.js

```diff
@@ -26,3 +27,59 @@ tour.register("carousel_content_removal", {
     content: "Check for a carousel slide with an empty container tag",
     run: function () {},
 }]);
+
+tour.register("snippet_carousel", {
```

Comment:
Ideally, this commit should be the first one, and each individual commit that follow filled the test with what was not working. But maybe nothing was really working/was testable before? ^^ Anyways, let's not waste time on this nitpick, it's already nice we now have a test :clap: However: mention in the previous commit message that a test will be added in the last commit. That way if we happen to fall on one of those commits in a few years, we'll know :+1:

Reply:
Nothing was broken before in terms of carousel options (like if we added a slide, the slide was added) so the test would work as a first commit 
I added it at the end because I wanted to make a better test testing the undo too => it would have been broken as a first commit
But when trying to do it, the tour was going too fast to properly see each undo step, like the check step was passing because the classes are correct but visually, the slide was not visible directly so in the end I dropped this idea... Maybe I should still add it ?

Reply:
> Maybe I should still add it ?

Not sure I understood :grimacing: 

- If your full test was working before your PR, that's ok to add it in any place. Explain that in the commit message though.
- If part of your test was not working before your PR, then, ideally (but not required now), it would be nice that this part was added at the same time (same commit) as the fix
- Your PR seems to fix things that could be tested (e.g. the first one seems quite easy to test, no?) => the test could be modified in that specific commit (if the test was added as a first commit)

Reply:
> Not sure I understood

Never mind, I will improve that  :+1:

Reply:
I added some steps at the end that slides the carousel on a slide that is not the first one, then save and then checks if the first slide is active :+1: 
I also added in the concerned commit messages that a test is added in the last commit :slightly_smiling_face: